src/shared/components/app/app.tsx

```diff
@@ -29,96 +125,17 @@ export class App extends Component<any, any> {
-            )}
-            <Navbar siteRes={siteRes} />
-            {siteRes?.site_view.site.content_warning && (
-              <AdultConsentModal
-                contentWarning={siteRes.site_view.site.content_warning}
-              />
-            )}
-            <div className="mt-4 p-0 fl-1">
-              <Switch>
-                {routes.map(
-                  ({
-                    path,
-                    component: RouteComponent,
-                    fetchInitialData,
-                    getQueryParams,
-                  }) => (
-                    <Route
-                      key={path}
-                      path={path}
-                      exact
-                      component={routeProps => {
-                        if (!fetchInitialData) {
-                          FirstLoadService.falsify();
-                        }
-
-                        let queryProps = routeProps;
-                        if (getQueryParams && this.isoData.site_res) {
-                          // ErrorGuard will not render its children when
-                          // site_res is missing, this guarantees that props
-                          // will always contain the query params.
-                          queryProps = {
-                            ...routeProps,
-                            ...getQueryParams(
-                              routeProps.location.search,
-                              this.isoData.site_res,
-                            ),
-                          };
-                        }
-
-                        return (
-                          <ErrorGuard>
-                            <div tabIndex={-1}>
-                              {RouteComponent &&
-                                (isAuthPath(path ?? "") ? (
-                                  <AuthGuard {...routeProps}>
-                                    <RouteComponent {...queryProps} />
-                                  </AuthGuard>
-                                ) : isAnonymousPath(path ?? "") ? (
-                                  <AnonymousGuard>
-                                    <RouteComponent {...queryProps} />
-                                  </AnonymousGuard>
-                                ) : (
-                                  <RouteComponent {...queryProps} />
-                                ))}
-                            </div>
-                          </ErrorGuard>
-                        );
-                      }}
-                    />
-                  ),
-                )}
-                <Route component={ErrorPage} />
-              </Switch>
-            </div>
-            <Footer site={siteRes} />
-          </div>
-        </Provider>
-      </>
+      <Provider i18next={I18NextService.i18n}>
+        {contentWarning && (
+          <AdultConsentModal contentWarning={contentWarning} />
+        )}
+        <AppForwardRef ref={this.rootRef} />
```

Comment:
Why not just set an `invisible` class depending on the contentWarning?

Reply:
Currently I handle the blur by setting the `background-filter` for the top level of the modal element. I've tried doing it by putting a class on app, but that always ended up blurring out the modal as well. Another approach I tried was just having all of the lifecycle methods and event handlers be handled in the original top-level `App` component. This seemed to work, but would cause the loading skeleton to flash in the blurred background each second the redirect time counts down.

I'm going to see if I can handle this all at the very top level `componentDidRoute`. I strongly suspect the issue is due to the order rendering is done in. Before my most recent change, the logic for showing the modal didn't run  until after the top level `App` component mounted and rendered. In the current iteration, `componentDidMount` for the consent modal is being called simultaneously with the `componentDidMount` for the `App` component. I think moving this logic to the very root of the component tree as a parent of `App` should solve the issue.

Reply:
Cool. If all else fails we might have to just resort to a simple confirm dialog on the SSR like lemmynsfw does it.

Reply:
> In the current iteration, `componentDidMount` for the consent modal is being called simultaneously with the `componentDidMount` for the `App` component.

`setState` in `componentWillMount` will affect the first render of the component. It gets called on the server too.

To avoid ever showing the wrongly un/blurred version after a page refresh, you'd probably need a `create-ssr-html`-hack similar to the auto detected light/dark themes. It's the same problem, the server doesn't know the real style to use, and whatever the server decides to show will otherwise be visible at least until `startClient` is called.

https://github.com/LemmyNet/lemmy-ui/blob/accf1b2d72c48f9b4eb0ee265bf6098a84079813/src/server/utils/create-ssr-html.tsx#L82-L87

Reply:
I've tried the cookie approach with and without a data attribute. The HTML and CSS that gets returned from the server before client side JS runs still isn't blurred. I think the problem is that bootstrap modals depend on being run in the browser to render. We'll either have to live with a slight delay for when the blur starts or go back to using the regular confirm dialog.